src/main/java/org/reaktivity/nukleus/kafka/internal/memory/MemoryLayout.java

```diff
@@ -42,63 +41,61 @@
     public static final int BITS_PER_BTREE_NODE = 1 << BITS_PER_BTREE_NODE_SHIFT;
     public static final int MASK_PER_BTREE_NODE = (1 << BITS_PER_BTREE_NODE) - 1;
 
-    public static final int LOCK_OFFSET = 0;
-    public static final int LOCK_SIZE = Long.BYTES;
-    public static final int MINIMUM_BLOCK_SIZE_OFFSET = LOCK_OFFSET + LOCK_SIZE;
+    public static final int MINIMUM_BLOCK_SIZE_OFFSET = 0;
     public static final int MINIMUM_BLOCK_SIZE_SIZE = Integer.BYTES;
     public static final int MAXIMUM_BLOCK_SIZE_OFFSET = MINIMUM_BLOCK_SIZE_OFFSET + MINIMUM_BLOCK_SIZE_SIZE;
-    public static final int MAXIMUM_BLOCK_SIZE_SIZE = Integer.BYTES;
+    public static final int MAXIMUM_BLOCK_SIZE_SIZE = Long.BYTES;
     public static final int BTREE_OFFSET = MAXIMUM_BLOCK_SIZE_OFFSET + MAXIMUM_BLOCK_SIZE_SIZE;
 
+    public static final long MAX_MAPPABLE_BYTES = Integer.MAX_VALUE;
+    public static final int ONE_GB = 0x40000000;
+
     private final AtomicBuffer metadataBuffer;
-    private final MutableDirectBuffer memoryBuffer;
+    private final MutableDirectBuffer[] memoryBuffers;
 
     private MemoryLayout(
         AtomicBuffer metadataBuffer,
-        MutableDirectBuffer memoryBuffer)
+        MutableDirectBuffer[] memoryBuffers)
     {
         this.metadataBuffer = metadataBuffer;
-        this.memoryBuffer = memoryBuffer;
+        this.memoryBuffers = memoryBuffers;
     }
 
     @Override
     public void close()
     {
         unmap(metadataBuffer.byteBuffer());
-        unmap(memoryBuffer.byteBuffer());
+        for (MutableDirectBuffer memoryBuffer : memoryBuffers)
+        {
+            unmap(memoryBuffer.byteBuffer());
+        }
     }
 
     public AtomicBuffer metadataBuffer()
     {
         return metadataBuffer;
     }
 
-    public MutableDirectBuffer memoryBuffer()
+    public MutableDirectBuffer[] memoryBuffers()
     {
-        return memoryBuffer;
+        return memoryBuffers;
     }
 
     public int minimumBlockSize()
     {
         return metadataBuffer.getInt(MINIMUM_BLOCK_SIZE_OFFSET);
     }
 
-    public int maximumBlockSize()
-    {
-        return metadataBuffer.getInt(MAXIMUM_BLOCK_SIZE_OFFSET);
-    }
-
-    public int capacity()
+    public long maximumBlockSize()
     {
-        return memoryBuffer.capacity();
+        return metadataBuffer.getLong(MAXIMUM_BLOCK_SIZE_OFFSET);
     }
 
     public static final class Builder extends Layout.Builder<MemoryLayout>
     {
         private Path path;
-        private int capacity;
         private int minimumBlockSize;
-        private int maximumBlockSize;
+        private long maximumBlockSize;
```

Comment:
`maximumBlockSize` and `capacity` are independent concepts, where `maximumBlockSize <= capacity` in general, and in this new implementation `maximumBlockSize <= 1GB` and `capacity <= Integer.MAX_VALUE * 1GB`.

The initial implementation just made a simplifying assumption that `maximumBlockSize == capacity`, but that is not generally accurate. Some of the changes in this PR are locking in that assumption, but it seems like a good opportunity to properly split the concepts, since `capacity` can be greater than `1GB` but `maximumBlockSize <= 1GB` now.

Reply:
I was assuming that one day the two concepts will be unified again. But I  would be happy to rename maximumBlockSize to capacity to make  the current meaning clearer.